### PR TITLE
Remove guava from datastore

### PR DESCRIPTION
**Pull request by elefeint**

### CODE REVIEW

**Comment by elefeint** [comment]
@dmitry-s Could you take a look at `DefaultDatastoreEntityConverterTests.testCollectionFieldsUnsupportedCollection`? It's failing now (because the code under test fails to fail). I think it's because while converting from `ArrayList` to `ImmutableSet` is invalid, converting from `ArrayList` to `Set` is just fine.

File: spring-cloud-gcp-data-datastore/src/test/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQueryTests.java [hunk 1]
```diff
@@ -533,7 +536,9 @@ private void prepareSliceResults(int offset, Integer queryLimit, Integer resultL
 					: results;
 		});
 		when(this.datastoreTemplate.convertEntitiesForRead(any(), any())).then(
-				(invocation) -> Lists.newArrayList(invocation.<Iterator>getArgument(0))
+				(invocation) -> StreamSupport.stream(Spliterators.spliteratorUnknownSize(
```
Comment (meltsufin) [inline spring-cloud-gcp-data-datastore/src/test/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQueryTests.java]: Using `ArrayList.addAll()` feels a bit less risky than the stream approach.

File: spring-cloud-gcp-data-datastore/src/test/java/org/springframework/cloud/gcp/data/datastore/core/convert/DefaultDatastoreEntityConverterTests.java [hunk 2]
```diff
@@ -301,45 +303,66 @@ public void testUnsupportedTypeWrite() {
...
 				"org.springframework.cloud.gcp.data.datastore.core.convert.TestDatastoreItemCollections entity;");
-		this.thrown.expectMessage("Unable to read property doubleSet;");
+		this.thrown.expectMessage("Unable to read property beanContext;");
+
 		this.thrown.expectMessage(
 				"Failed to convert from type [java.util.ArrayList<?>] " +
-						"to type [com.google.common.collect.ImmutableSet<?>]");
+						"to type [org.springframework.cloud.gcp.data.datastore.core.convert.TestDatastoreItemCollections$ComparableBeanContextSupport]");
+
+		ComparableBeanContextSupport ComparableBeanContextSupport = new ComparableBeanContextSupport();
```
Comment (artembilan) [inline spring-cloud-gcp-data-datastore/src/test/java/org/springframework/cloud/gcp/data/datastore/core/convert/DefaultDatastoreEntityConverterTests.java]: Typical Java code style to have classes capitalized, but variables in camel case starting with non-capitalized letter.

**Comment by elefeint** [comment]
Part of #1198

**Pull request merged**